connectors/tests/test_job_scheduling.py

```diff
@@ -266,3 +266,26 @@ async def test_connector_prepare_failed(
     connector.heartbeat.assert_not_awaited()
     connector.reset_sync_now_flag.assert_not_awaited()
     concurrent_tasks_mock.put.assert_not_awaited()
+
+
+@pytest.mark.asyncio
+async def test_run_when_sync_fails_then_continues_service_execution(
+    connector_index_mock, concurrent_tasks_mock, set_env
+):
+    connector = mock_connector(sync_now=True, next_sync=0)
+    connector_index_mock.supported_connectors.return_value = AsyncIterator([connector])
+
+    # seems easier to mock this
+    concurrent_tasks_mock.put.side_effect = Exception("Something went wrong!")
```

Comment:
This will soon be moved to `JobExecutionService` in #593

https://github.com/elastic/connectors-python/blob/3cfabc3f0b366da7ab900042c752525f8b3ad65f/connectors/services/job_scheduling.py#L119

Maybe you can consider using

```python
connector.heartbeat.side_effect = Exception("Something went wrong!")
```

Reply:
Sure, I'll change it to `heartbeat` for simplicity of migration